src/main/java/uk/gov/ida/verifyserviceprovider/utils/ServiceEntityIdHelper.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.ida.verifyserviceprovider.utils;
```

Comment:
I'm not convinced at all that this is a Helper, or that it should be in utils. (It morphed a bit after first writing).

Opinions on renaming/moving this class are particularly sought.

Reply:
As are ones on rewording log messages. (Especially on rewording them in a way which would make this able to have one function, not two)

Reply:
Seems like this whole class is a bit confused? It is a utility to abstract out the case of single entityId vs list of entityIds, but it is also doing validation that the entityId is valid. I think with it's current name, it is not clear what it is doing.

I would prefer this to just be a validator then we just do `validate(entityId)` line then just do an unchecked access to entityId afterwards (as we know it is valid if it made it past the validate line)

At the very least we can commonise the else block and flip the condition so that the single entityId case (what we expect to be normal) comes first (small processing optimisation?). Maybe make the 2 dtos inherit an interface that says they have an entityId on them? Don't know how is best to handle error messages then but it would bring the logic in line.

Reply:
Error messages was my thinking - otherwise I would pass in the potentially null entity Id to a single function.